docs/tutorial/object_decomp.md

```diff
@@ -1,14 +1,160 @@
+- Animations
+- Texture Animations (instructions to scroll/cycle textures or change colors)
+
+## How we work with objects
+
+Because these are regarded as copyrighted, we do not want them in the repository. We instead extract them from the ROM. To do this we use a system called ZAPD (Zelda Asset Processor for Decompilation). The main aim of object decompilation is to give ZAPD an XML file that tells it what the assets it is supposed to be extracting actually are, so it can put them in the right format and name them.
+
+## How to decomp an object
+
+Choose an object to decomp. As usual, some will be easier than others. For reasons explained shortly, it is much easier to decomp an object if all actors that use it are decompiled.
+
+### Files and folders
+
+Select the XML file of your selected object, which should be in `assets/xml/objects/object_name.xml`.
+
+The ZAPD output will go in the folder `assets/objects/object_name/`. You'll want this folder open later to check the output is correct.
+
+### Examining actor files
+
+All objects have had their XML generated in an automated way, so most constituent parts of each object are already identified, but will still need to be named and documented properly. Further, these objects usually have some blobs unreferenced by the object's own contents and hence not automatically extracted; most of the time these can be identified by looking at references in the actor which uses said object.
+
+### Extracting assets
+
+You can run `extract_assets.py` to extract the object's assets. Running it with no arguments extracts all files that have been updated since the last time it was run. You can also run it with `-s` (for single file), and give it the location of the object you want to extract relative to `assets`, i.e.
+
+```bash
+./extract_assets.py -s objects/object_name
+```
+
+This should populate the folder you created earlier. ZAPD produces a C file containing the extracted object data, which will be `assets/objects/object_name/object_name.c`. Any data that you have not specified the type of, or is not referenced elsewhere in the object, is extracted as unknown blobs  (usually named `unaccounted_XXXXXX`).  Open the C file to see if there are any such blobs. (Some are just padding with 0s and can be ignored.)
+
+You now have to try and decipher these blobs using the [list of tools given below](#tools) to work out what they might be. In the case of unused parts of the object, this can be very difficult.
+
+### Naming
+
+You'll want to name everything that exists in the XML, but it helps to break it down into separate steps. After each step, extract the single asset again.
+
+1. First, name every piece of data that is directly accessed by another part of the repo (generally, but not always, actors). This allows for elimination of entries in [`undefined_syms.txt`](../../undefined_syms.txt). For most objects, this includes several of a skeleton, animations, some textures, and collision information.
```

Comment:
Come to think of it, we should probably add a small part to the code decomp tutorial of including the object file and using the autogenned names. Not sure where it would go though.

Reply:
Yeah, I'm not sure where to put it either. I'll just delete this sentence for now and if someone thinks of a good place to stick it, I'll include it there.